backends/p4tools/modules/testgen/testgen.cpp

```diff
@@ -91,12 +90,12 @@ int generateAbstractTests(const TestgenOptions &testgenOptions, const ProgramInf
     return ::errorCount() == 0 ? EXIT_SUCCESS : EXIT_FAILURE;
 }
 
-int Testgen::mainImpl(const IR::P4Program *program) {
+int Testgen::mainImpl(const CompilerResult &compilerResult) {
     // Register all available testgen targets.
     // These are discovered by CMAKE, which fills out the register.h.in file.
     registerTestgenTargets();
 
-    const auto *programInfo = TestgenTarget::initProgram(program);
+    const auto *programInfo = TestgenTarget::initProgram(&compilerResult.getProgram());
```

Comment:
I presume the next step would be to actually pass the `compilerResult` to `TestgenTarget::initProgram`. Would it make sense to do this change already now? Without that, this does not actually change anything...

Reply:
I initially had all changes bundled up in one PR, but decided to split the PR because the amount files touched grew quite large. #4324 alone touches ~30 files.